pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned_test.go

```diff
@@ -112,11 +114,50 @@ var _ = Describe("Tuned", func() {
 			})
 
 			It("should contain high power consumption related parameters", func() {
+				profile.Spec.WorkloadHints.PerPodPowerManagement = pointer.BoolPtr(false)
 				manifest := getTunedManifest(profile)
 				By("Populating high power consumption cmdline")
 				Expect(cmdlineHighPowerConsumption.MatchString(manifest)).To(BeTrue())
 				By("Populating idle poll cmdline")
 				Expect(cmdlineHighPowerConsumptionWithRealtime.MatchString(manifest)).To(BeTrue())
+				By("Populating pstate cmdline")
+				Expect(cmdlineHighPowerConsumptionWithoutPerPodPowerManagement.MatchString(manifest)).To(BeTrue())
+			})
+		})
+
+		When("additional args", func() {
+			It("should contain additional additional parameters", func() {
+				manifest := getTunedManifest(profile)
+				Expect(manifest).To(ContainSubstring("audit=0"))
+				Expect(manifest).To(ContainSubstring("processor.max_cstate=1"))
+				Expect(manifest).To(ContainSubstring("idle=poll"))
+				Expect(manifest).To(ContainSubstring("intel_idle.max_cstate=0"))
+			})
+
+			It("should not contain additional additional parameters", func() {
+				profile.Spec.AdditionalKernelArgs = nil
+				manifest := getTunedManifest(profile)
+				Expect(manifest).ToNot(ContainSubstring("audit=0"))
+				Expect(manifest).ToNot(ContainSubstring("processor.max_cstate=1"))
+				Expect(manifest).ToNot(ContainSubstring("idle=poll"))
+				Expect(manifest).ToNot(ContainSubstring("intel_idle.max_cstate=0"))
+			})
+		})
+
+		When("perPodPowerManagement Hint to false", func() {
+			It("should not contain perPodPowerManagement related parameters", func() {
+				manifest := getTunedManifest(profile)
+				Expect(manifest).ToNot(ContainSubstring("enabled=false"))
+				Expect(cmdlinePerPodPowerManagementHint.MatchString(manifest)).ToNot(BeTrue())
+			})
+		})
+
+		When("perPodPowerManagement Hint to true", func() {
+			It("should contain perPodPowerManagement related parameters", func() {
+				profile.Spec.WorkloadHints.PerPodPowerManagement = pointer.BoolPtr(true)
+				manifest := getTunedManifest(profile)
+				Expect(manifest).To(ContainSubstring("enabled=false"))
```

Comment:
This is too generic I think, we can use enabled/disabled in other parts of the profile too

Reply:
Add a regex, not very happy with it but I don't see other option

Reply:
Fixed here: https://github.com/openshift/cluster-node-tuning-operator/pull/426